src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CheckForPoison.cs

```diff
@@ -370,11 +368,18 @@ private static bool IsAttributeSbrp(MetadataReader reader, CustomAttribute attr)
 
             if (attributeType == SbrpAttributeType)
             {
-                BlobReader blobReader = reader.GetBlobReader(attr.Value);
-                string attributeValue = Encoding.UTF8.GetString(blobReader.ReadBytes(blobReader.Length));
-                attributeValue = Regex.Replace(attributeValue, @"\p{C}+", string.Empty);
-                return Regex.IsMatch(attributeValue, SbrpAttributeValuePattern);
+                var provider = new CustomAttributeTypeProvider(reader);
+                var decodedValue = attr.DecodeValue(provider);
+                try
+                {
+                    return decodedValue.FixedArguments[0].Value.ToString() == "source" && decodedValue.FixedArguments[1].Value.ToString() == "source-build-reference-packages";
+                }
+                catch
+                {
+                    throw new Exception($"{SbrpAttributeType} is not formatted properly with a key, value pair.");
```

Comment:
[Design guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/using-standard-exception-types#exception-and-systemexception) suggest never throwing the base exception type.  InvalidOperationException seems more appropriate.